src/marvin/utilities/pydantic.py

```diff
@@ -1,8 +1,9 @@
+"""Module for Pydantic utilities."""
 from types import FunctionType, GenericAlias
 from typing import Annotated, Any, Callable, Optional, Union, cast, get_origin
 
 from pydantic import BaseModel, TypeAdapter, create_model
-from pydantic.v1 import validate_arguments
+from pydantic.deprecated.decorator import validate_arguments
```

Comment:
because we were using `v1`'s `validate_arguments`, we were getting v1 `BaseModel` instances here as well.

probably not ideal to use this (being deprecated) and should switch to `validate_call`  (although I couldn't figure out how to get the model back from that right away, may need `TypeAdapter` for this?)